query-node/mappings/src/content/channel.ts

```diff
@@ -77,13 +85,37 @@ export async function content_ChannelCreated(ctx: EventContext & StoreContext):
       inconsistentState(`storageBag for channel ${channelId} does not exist`)
     }
 
-    const metadata = deserializeMetadata(ChannelMetadata, channelCreationParameters.meta.unwrap()) || {}
-    await processChannelMetadata(ctx, channel, metadata, dataObjects)
+    const appAction = deserializeMetadata(AppAction, channelCreationParameters.meta.unwrap())
+
+    if (appAction && Object.keys(appAction).length) {
```

Comment:
I noticed there is a backward-compatibility problem both here and in the `CreateVideo` event handler.
The problem is that in protobuf you cannot distinguish between different message types (like `AppAction` vs `ChannelMetadata`) just by looking at the serialized message.

For example, a message like:
```
ChannelMetadata {
  title: "Test",
  description: "Test"
}
```
Will be exactly the same as:
```
AppAction {
  appId: "Test",
  metadata: "Test"
}
```
once serialized. Both of them will be serialized to `0x0a0454657374120454657374`.

You can only distringuish a serialized `AppAction` from `ChannelMetadata` if a property which has an index that exists only in one of the messages (for example, property `6` only exists in `ChannelMetadata`) is specified, or if a property which has different type depending on the message (for example, `3` is `bool` in `ChannelMetadata`, but `string` in `AppAction`) is encoded in such a way that you can derive the type from the encoding.

So what we need is a way to be able to definitely tell whether a serialized message was supposed to be `AppAction` or `ChannelMetadata` / `ContentMetadata`. You can require a prefix for `AppAction` messages for example, or make `app_id` a `required` property with some unique index, for example:
```
    message AppAction {
      required string app_id = 9999;
    }
```